scripts/extract_metadata.py

```diff
@@ -302,34 +302,52 @@ def parse_thumbnails(self):
             os.path.basename(self.path))[0] + ".png"
         thumbPath = os.path.join(
             os.path.dirname(self.path), "thumbs", thumbName)
-        if not os.path.isfile(thumbPath):
-            return None
-        # read file
+        thumb_matches = re.findall(
+            r"; thumbnail begin[;/\+=\w\s]+?; thumbnail end", self.header_data)
         thumbs = []
-        try:
-            with open(thumbPath, 'rb') as thumbFile:
-                fbytes = thumbFile.read()
-                with Image.open(io.BytesIO(fbytes)) as im:
-                    thumbFull = base64.b64encode(fbytes).decode()
-                    thumbs.append({
-                        'width': im.width, 'height': im.height,
-                        'size': len(thumbFull), 'data': thumbFull
-                    })
-                    # Create 32x32 thumbnail
-                    im.thumbnail((32, 32), Image.ANTIALIAS)
-                    tmpThumb = io.BytesIO()
-                    im.save(tmpThumb, format="PNG")
-                    thumbSmall = base64.b64encode(
-                        tmpThumb.getbuffer()).decode()
-                    tmpThumb.close()
-                    thumbs.insert(0, {
-                        'width': im.width, 'height': im.height,
-                        'size': len(thumbSmall), 'data': thumbSmall
-                    })
-        except Exception as e:
-            self.log.append(str(e))
-            return None
-        return thumbs
+        if os.path.isfile(thumbPath):
+            # read file
+            try:
+                with open(thumbPath, 'rb') as thumbFile:
+                    fbytes = thumbFile.read()
+                    with Image.open(io.BytesIO(fbytes)) as im:
+                        thumbFull = base64.b64encode(fbytes).decode()
+                        thumbs.append({
+                            'width': im.width, 'height': im.height,
+                            'size': len(thumbFull), 'data': thumbFull
+                        })
+                        # Create 32x32 thumbnail
+                        im.thumbnail((32, 32), Image.ANTIALIAS)
```

Comment:
Is there a reason these are hard coded?

32x32, on a retina display - effectively means 16x16 if you want to keep clarity.

If there's a true requirement to hard code dimensions here, then maybe a better approach would be to create 64x64 - or some variation of.

Reply:
I'm sorry for responding so late! I didn't see this. The reason this is here is for backwards compatibility with the original code. I wasn't sure why these numbers in particular were chosen but I didn't want to modify existing implementation.
Everything from ```if thumb_matches:``` and below is the new code

Reply:
It is hardcoded because I can be sure it is functional.  32x32 is simple the size for the smaller thumbnail, the idea was to be compatible with the Prusa defaults and generate a thumbnail that works well with low resolution screens.  I think if a client want 64x64 they could simply scale the 300x300 thumbnail to whatever size they need.

Reply:
Is this not configurable user side? Does it just generate 32x32 and a 300x300?